core/src/main/java/com/radixdlt/consensus/ProposalLimitsConfig.java

```diff
@@ -0,0 +1,128 @@
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.consensus;
+
+import com.google.inject.AbstractModule;
+import com.radixdlt.statemanager.VertexLimitsConfig;
+
+/**
+ * [`maxTransactionCount`]: Maximum number of transactions to include in a proposal.
+ * [`maxTransactionsPayloadSize`]: Maximum number of transaction payload bytes to include in a
+ * proposal. [`maxUncommittedTransactionsPayloadSize`]: Maximum number of transaction payload bytes
+ * to include in a proposal and its previous vertices chain. Intended to limit the size of a commit
+ * batch (i.e. the size of transactions under a single commit proof). Note - we can still keep
+ * committing round changes, so this is not a guarantee. But should be reasonably effective as a
+ * limit.
+ */
+public record ProposalLimitsConfig(
+    int maxTransactionCount,
+    int maxTransactionsPayloadSize,
+    int maxUncommittedTransactionsPayloadSize) {
+  public static final int DEFAULT_MAX_TRANSACTION_COUNT = 4;
+  public static final int DEFAULT_MAX_TRANSACTIONS_PAYLOAD_SIZE = 2 * 1024 * 1024;
+  public static final int DEFAULT_MAX_UNCOMMITTED_TRANSACTIONS_PAYLOAD_SIZE = 2 * 1024 * 1024;
+
+  public static ProposalLimitsConfig zero() {
+    return new ProposalLimitsConfig(0, 0, 0);
+  }
+
+  public static ProposalLimitsConfig singleTransaction() {
+    return new ProposalLimitsConfig(1, 1024 * 1024, 5 * 1024 * 1024);
+  }
+
+  public static ProposalLimitsConfig testDefaults() {
+    return new ProposalLimitsConfig(10, 10 * 1024 * 1024, 50 * 1024 * 1024);
+  }
+
+  public static ProposalLimitsConfig defaults() {
+    return new ProposalLimitsConfig(
+        DEFAULT_MAX_TRANSACTION_COUNT,
+        DEFAULT_MAX_TRANSACTIONS_PAYLOAD_SIZE,
+        DEFAULT_MAX_UNCOMMITTED_TRANSACTIONS_PAYLOAD_SIZE);
+  }
+
+  public static ProposalLimitsConfig from(VertexLimitsConfig vertexLimitsConfig) {
+    return new ProposalLimitsConfig(
+        // we add an extra transaction to build the vertex from
+        vertexLimitsConfig.maxTransactionCount().toInt() + 1,
+        // even though we would like to fill the vertex transaction count to the maximum possible,
+        // we want to keep
+        // bandwidth strict
+        vertexLimitsConfig.maxTotalTransactionsSize().toInt(),
+        // we multiply by 3 for the 3-chain rule
+        vertexLimitsConfig.maxTotalTransactionsSize().toInt() * 3);
```

Comment:
these lines look like `ProposalLimitsConfig` is actually computable from `VertexLimitsConfig` in a very reasonable way - do we still need to be able to specify the `ProposalLimitsConfig` independently?

Reply:
I was also thinking exposing proposal limits as well to tests different cases but this is second priority and we can do after a first round of tuning.